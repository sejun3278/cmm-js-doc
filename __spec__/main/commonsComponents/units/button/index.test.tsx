import "@testing-library/jest-dom";
import { render } from "@testing-library/react";

import _Button from "src/main/commonsComponents/units/button";
import { useRouter } from "next/router";

// 가짜 router 만들기
jest.mock("next/router", () => ({
  useRouter: jest.fn(),
}));
const push = jest.fn();
(useRouter as jest.Mock).mockImplementation(() => ({
  push,
}));

export default it("Button Units Page Sanpshot", () => {
  const component = render(
    <_Button onClickEvent={() => console.log("버튼 클릭")} className="jest_btn">
      버튼 테스트
    </_Button>
  );
  // @ts-ignore
  expect(component.container).toMatchSnapshot();
});
