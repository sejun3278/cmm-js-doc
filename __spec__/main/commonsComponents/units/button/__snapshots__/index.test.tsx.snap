// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Units Page Jest Button Units Page - Snapshot 1`] = `
<div>
  <button
    class="_button_ jest_btn"
    role="button_click_event"
  >
    버튼 테스트
  </button>
</div>
`;

exports[`Button Units Page Jest Button Units Page Snapshot 1`] = `
<div>
  <button
    class="_button_ jest_btn"
    role="button_click_event"
  >
    버튼 테스트
  </button>
</div>
`;
